packages/caravan-fees/src/rbf.ts

```diff
@@ -0,0 +1,514 @@
+    }
+  }
+
+  // getters
+
+  get psbt(): PsbtV2 {
+    return this.newPsbt;
+  }
+
+  get targetFeeRate(): FeeRateSatsPerVByte {
+    return this._targetFeeRate;
+  }
+
+  get totalInputValue(): string {
+    return calculateTotalInputValue(this.newPsbt).toString();
+  }
+
+  get totalOutputValue(): string {
+    return calculateTotalOutputValue(this.newPsbt).toString();
+  }
+
+  get currentFee(): string {
+    return new BigNumber(this.totalInputValue)
+      .minus(this.totalOutputValue)
+      .toString();
+  }
+
+  get requiredFee(): string {
+    return new BigNumber(this.estimateVsize())
+      .multipliedBy(this.targetFeeRate)
+      .toString();
+  }
+
+  get feeIncrease(): string {
+    return new BigNumber(this.requiredFee).minus(this.currentFee).toString();
+  }
+
+  // Methods
+  /**
+   * Sets a new target fee rate for the RBF transaction.
+   *
+   * @param {FeeRateSatsPerVByte} feeRate - The new target fee rate
+   * @returns {RbfTransaction} A new instance with the updated fee rate
+   */
+  public setTargetFeeRate(feeRate: FeeRateSatsPerVByte): RbfTransaction {
+    //The spread operator { ...this } doesn't work as intended in this context because
+    // the RbfTransaction instance doesn't have all the properties required by RbfTransactionOptions
+    const newInstance = new RbfTransaction({
+      psbt: this.originalPsbt,
+      network: this.network,
+      targetFeeRate: feeRate,
+      feeOutputIndex: this.feeOutputIndex,
+      dustThreshold: this.dustThreshold.toNumber(),
+      additionalUtxos: this.additionalUtxos,
+      requiredSigners: this.requiredSigners,
+      totalSigners: this.totalSigners,
+      changeOutputIndices: this.changeOutputIndices,
+    });
+    newInstance._targetFeeRate = feeRate;
+    return newInstance;
+  }
+
+  public getFinalPsbt(): PsbtV2 {
+    return this.newPsbt;
+  }
+
+  /**
+   * Prepares an accelerated replacement transaction
+   *
+   * This method creates a new PSBT with increased fees to accelerate confirmation.
+   * It adheres to BIP125 rules and adjusts outputs or adds inputs as necessary.
+   *
+   * @returns {RbfTransaction} A new instance with the accelerated transaction
+   * @throws {Error} If unable to create a valid replacement transaction
+   */
+  public prepareAccelerated(): RbfTransaction {
+    const newInstance = new RbfTransaction({
+      psbt: this.originalPsbt,
+      network: this.network,
+      targetFeeRate: this.targetFeeRate,
+      feeOutputIndex: this.feeOutputIndex,
+      dustThreshold: this.dustThreshold.toNumber(),
+      additionalUtxos: this.additionalUtxos,
+      requiredSigners: this.requiredSigners,
+      totalSigners: this.totalSigners,
+      changeOutputIndices: this.changeOutputIndices,
+    });
+    newInstance.updateSequenceNumbers();
+    newInstance.adjustOutputs();
+    newInstance.addAdditionalInputsIfNeeded();
+    newInstance.updateFeeRate();
+    return newInstance;
```

Comment:
Again, this isn't right. If you're avoiding mutating the underlying transactions, I would make a copy of the underlying PSBT, make adjustments and return that updated PSBT (but preserve the base PSBT for later operations).

Reply:
Yes ... thanks :)

it's refactored [now](https://github.com/caravan-bitcoin/caravan/pull/114/commits/4f4f97a08243f63ca31c4f0fc34216671e97e1de#diff-47122bb461c82a41c84d8866adf0cb005bf7f573a2afb8558f670a419bed146aR197) 

added a new resetState method to handle the idempotency and to avoid mutations ... [here](https://github.com/caravan-bitcoin/caravan/pull/114/commits/4f4f97a08243f63ca31c4f0fc34216671e97e1de#diff-47122bb461c82a41c84d8866adf0cb005bf7f573a2afb8558f670a419bed146aR264)